### PR TITLE
Refactoring workflow update

### PR DESCRIPTION
**Pull request by jcoyne**
Add logging to StatusListService
Remove Permission Template logic from AdminSetsController

### CODE REVIEW

File: app/forms/hyrax/forms/admin_set_form.rb [hunk 1]
```diff
@@ -18,11 +17,8 @@ def [](key)
       end
 
       def permission_template
-        PermissionTemplateForm.new(@permission_template)
-      end
-
-      def workflow_name
-        @permission_template.workflow_name
+        template_model = PermissionTemplate.find_by!(admin_set_id: model.id)
+        PermissionTemplateForm.new(template_model)
```
Comment (jeremyf) [inline app/forms/hyrax/forms/admin_set_form.rb]: Is this something you want to memoize? And I noticed the `#workflow_name` method is being removed. Is that untested code?

File: app/services/hyrax/workflow/status_list_service.rb [hunk 2]
```diff
@@ -19,13 +19,20 @@ def each
 
       private
 
+        delegate :logger, to: :context
+
+        def user
```
Comment (jeremyf) [inline app/services/hyrax/workflow/status_list_service.rb]: The user method is privatized (which I consider a good thing in regards to methods) but it is a change to the interface of the object.

File: spec/controllers/hyrax/admin/admin_sets_controller_spec.rb [hunk 3]
```diff
@@ -113,16 +113,11 @@
 
     describe "#update" do
       let(:admin_set) { create(:admin_set, edit_users: [user]) }
-      let(:permission_template) { Hyrax::PermissionTemplate.find_or_create_by(admin_set_id: admin_set.id) }
       it 'updates a record' do
-        # Prevent a save which causes Fedora to complain it doesn't know the referenced node.
-        expect_any_instance_of(AdminSet).to receive(:save).and_return(true)
         patch :update, params: { id: admin_set,
-                                 admin_set: { title: "Improved title", thumbnail_id: "mw22v559x", workflow_name: "one_step_mediated_deposit" } }
+                                 admin_set: { title: "Improved title" } }
```
Comment (jeremyf) [inline spec/controllers/hyrax/admin/admin_sets_controller_spec.rb]: I'm not aware of the UI interaction. Please explain the removal of the workflow_name from the update.

File: app/forms/hyrax/forms/admin_set_form.rb (hunk 1 above)
Comment (jcoyne) [inline app/forms/hyrax/forms/admin_set_form.rb]: This is dead code. The view is no longer using it (this change happened in a different PR). I'm just cleaning up.

File: spec/controllers/hyrax/admin/admin_sets_controller_spec.rb (hunk 3 above)
Comment (jcoyne) [inline spec/controllers/hyrax/admin/admin_sets_controller_spec.rb]: The UI is no longer using this path. I'm cleaning up cruft.

Comment (jeremyf) [inline spec/controllers/hyrax/admin/admin_sets_controller_spec.rb]: :)

**Comment by mjgiarlo** [comment]
@jcoyne Excellent work!

**Pull request merged**